### PR TITLE
Expose configs under connector id

### PR DESCRIPTION
**Pull request by hugjobk**
- Expose configs under connector id

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -39,6 +40,7 @@ impl Config {
             kafka_url,
             kafka_env,
             proto_registry_host,
+            value: yaml,
```
Comment (rukolahasser) [inline src/config.rs:43]: You should only expose the value you need instead of the entire file.

**Review by rukolahasser — CHANGES_REQUESTED** [review]

File: src/config.rs (hunk 1 above)
Comment (hugjobk) [reply src/config.rs:43]: I made it private and only expose configs under connector id

File: src/config.rs [hunk 2]
```diff
@@ -9,6 +9,7 @@ pub struct Config {
     pub kafka_url: String,
     pub kafka_env: Env,
     pub proto_registry_host: String,
+    pub value: serde_yaml::Value,
```
Comment (rukolahasser) [inline src/config.rs:12]:
```suggestion
    pub sub_config: Value,
```

File: src/config.rs (hunk 1 above)
Comment (rukolahasser) [reply src/config.rs:43]: It should be removed

File: src/connector.rs [hunk 3]
```diff
@@ -16,10 +16,13 @@ pub struct Connector {
 
 impl Connector {
     pub fn new() -> Self {
-        let config = Config::init();
+        let mut config = Config::init();
         let manifest = Manifest::init();
         let id = Connector::id(&manifest, &config);
         let producer = Producer::new(&config.kafka_url, &id);
+        let sub_config = &config.value[id];
```
Comment (rukolahasser) [inline src/connector.rs:23]: Create a method like `build_sub_config(&self, id &str)` in the `Config` struct to build sub_config

File: src/config.rs [hunk 4]
```diff
@@ -9,6 +9,7 @@ pub struct Config {
     pub kafka_url: String,
     pub kafka_env: Env,
     pub proto_registry_host: String,
+    pub value: Value,
```
Comment (rukolahasser) [inline src/config.rs:12]: change it to `sub_config`

File: src/config.rs (hunk 1 above)
Comment (rukolahasser) [reply src/config.rs]: Don't set the value here

File: src/config.rs [hunk 5]
```diff
@@ -59,4 +61,8 @@ impl Config {
 
         yaml
     }
+
+    pub fn sub_config(&self, path: &str) -> Value {
```
Comment (rukolahasser) [inline src/config.rs:65]:
```suggestion
    pub fn build_sub_config(&self, path: &str) {
        let yaml = Config::read_from_file();
        self.sub_config = yaml[path]
    }
```

**Review by rukolahasser — APPROVED** [review]

**Pull request merged**